### PR TITLE
[BED-92] Add PTR support

### PR DESCRIPTION
**Pull request by csucu**
Valid check host implementation should support PTR, even though it is not recommended to use.

### CODE REVIEW

**Comment by dmotylev** [comment]
I'd consider using https://pkg.go.dev/net/netip package to check IP family and other logic. There is also https://pkg.go.dev/go4.org/netipx which is not part of golang, but made by the same team. Both packages are modern and suit very well for the task

File: printer/printer.go [hunk 1]
```diff
@@ -93,6 +93,13 @@ func (p *Printer) LookupTXTStrict(name string) ([]string, time.Duration, error)
 	return p.r.LookupTXTStrict(name)
 }
 
+func (p *Printer) LookupPTR(name string) ([]string, time.Duration, error) {
+	fmt.Fprintf(p.w, "%s  lookup(TXT) %s\n", strings.Repeat("  ", p.c), name)
```
Comment (dmotylev) [inline printer/printer.go]:
```suggestion
	fmt.Fprintf(p.w, "%s  lookup(PTR) %s\n", strings.Repeat("  ", p.c), name)
```

File: parser.go [hunk 2]
```diff
@@ -600,9 +600,56 @@ func (p *parser) parseExists(t *token) (bool, Result, time.Duration, error) {
...
+			return false, nil
+		})
+		if err != nil {
+			continue
+		}
+
+		if found {
+			return true, result, nil
+		}
+	}
```
Comment (dmotylev) [inline parser.go:650]: What if `MatchIP` return `err!=nil` for all `ptrs`?

**Review by dmotylev — COMMENTED** [review]
It seems resolvers in `redsfit/spf` do not convert IP to `in-addr.arpa.` form in LookupPTR implementation

File: parser.go (hunk 2 above)
Comment (csucu) [reply parser.go:650]: Then we return result Fail, at the end of the method (line 652)

**Comment by csucu** [comment]
> It seems resolvers in `redsfit/spf` do not convert IP to `in-addr.arpa.` form in LookupPTR implementation

I've added the correct handling to `resolver_miekg.go`. The resolver in `resolver_std.go` uses `net.LookupAddr` which converts the ip internally.

**Review by dmotylev — APPROVED** [review]

**Pull request merged**